Source/RapyutaSimulationPlugins/Private/Robots/RobotVehicle.cpp

```diff
@@ -2,38 +2,123 @@
 
 #include "Robots/RobotVehicle.h"
 
-#include "Drives/RobotVehicleMovementComponent.h"
+// rclUE
 #include "Msgs/ROS2TFMsg.h"
 #include "ROS2Node.h"
 
-void ARobotVehicle::InitializeMoveComponent()
+// RapyutaSimulationPlugins
+#include "Drives/RobotVehicleMovementComponent.h"
+#include "RRROS2GameMode.h"
+#include "Tools/ROS2Spawnable.h"
+#include "Tools/SimulationState.h"
+
+ARobotVehicle::ARobotVehicle()
+{
+    Initialize();
+}
+
+ARobotVehicle::ARobotVehicle(const FObjectInitializer& ObjectInitializer) : Super(ObjectInitializer)
 {
-    RobotVehicleMoveComponent = NewObject<URobotVehicleMovementComponent>(this, TEXT("RobotVehicleMoveComponent"));
+    Initialize();
 }
 
-void ARobotVehicle::Tick(float DeltaSeconds)
+void ARobotVehicle::Initialize()
 {
-    Super::Tick(DeltaSeconds);
+    SkeletalMeshComp = CreateDefaultSubobject<USkeletalMeshComponent>(*FString::Printf(TEXT("%s_SkeletalComp"), *GetName()));

```

Comment:
Well in a generic sense, since we have Physics turtlebot that don't but Kinematics do, also this would be in line with a skeletal meshed based imported robots in the future, so I add it here to avoid extra duplicated adding in BP actors.